### ISSUE TITLE
sqlite-orm build failure

### ISSUE THREAD

**Issue by quickiwiki** [microsoft/vcpkg#21996]
**Host Environment**
- OS: Windows/Linux
- Compiler: VS 2017/g++ (Ubuntu 8.4.0-1ubuntu1~18.04) 8.4.0

**To Reproduce**
`./vcpkg install sqlite-orm`

**Failure logs** 
```
/vcpkg/installed/x64-linux/include/sqlite_orm/sqlite_orm.h:10124:25: error: ‘sqlite_orm::internal::is_upsert_clause’ has not been declared
         using internal::is_upsert_clause;
                         ^~~~~~~~~~~~~~~~
/vcpkg/installed/x64-linux/include/sqlite_orm/sqlite_orm.h:10146:68: error: ‘is_upsert_clause’ was not declared in this scope
         constexpr int upsertClausesCount = count_tuple<args_tuple, is_upsert_clause>::value;
                                                                    ^~~~~~~~~~~~~~~~
```
**Additional context**
The previous version of the lib (e.g.: sqlite-orm:x64-linux 1.6) worked correctly

**Comment by JackBoosY** [comment]
Can you please provide the following logs:
_buildtrees/sqlite-orm/config-x64-linux-dbg-out.log_
_buildtrees/sqlite-orm/install-x64-linux-dbg-out.log_

Thanks.

**Comment by quickiwiki** [comment]
Thanks, this happens when I try to use sqlite-orm in a project. I may have submitted to the wrong category

cat ~/vcpkg/buildtrees/sqlite-orm/config-x64-linux-dbg-out.log
-- The C compiler identification is GNU 8.4.0
-- The CXX compiler identification is GNU 8.4.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- SQLITE_ORM: Build with C++14 features
-- Configuring sqlite_orm 1.6.0
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- CMAKE_BUILD_TYPE: Debug
-- Configuring done
-- Generating done
-- Build files have been written to: /home/el/vcpkg/buildtrees/sqlite-orm/x64-linux-dbg

 cat ~/vcpkg/buildtrees/sqlite-orm/install-x64-linux-dbg-out.log
[0/1] cd /home/el/vcpkg/buildtrees/sqlite-orm/x64-linux-dbg && /home/el/vcpkg/downloads/tools/cmake-3.21.1-linux/cmake-3.21.1-linux-x86_64/bin/cmake -P cmake_install.cmake
-- Install configuration: "Debug"
-- Installing: /home/el/vcpkg/packages/sqlite-orm_x64-linux/debug/lib/cmake/SqliteOrm/SqliteOrmConfig.cmake
-- Installing: /home/el/vcpkg/packages/sqlite-orm_x64-linux/debug/lib/cmake/SqliteOrm/SqliteOrmConfigVersion.cmake
-- Installing: /home/el/vcpkg/packages/sqlite-orm_x64-linux/debug/include/sqlite_orm
-- Installing: /home/el/vcpkg/packages/sqlite-orm_x64-linux/debug/include/sqlite_orm/sqlite_orm.h
-- Installing: /home/el/vcpkg/packages/sqlite-orm_x64-linux/debug/lib/cmake/SqliteOrm/SqliteOrmTargets.cmake

**Comment by JackBoosY** [comment]
Okay, can you also please provide a sample code to reproduce this?

Thanks.

**Comment by JackBoosY** [comment]
Ping @quickiwiki  for response.

**Comment by quickiwiki** [comment]
Yes, sorry @JackBoosY I'll provide soon the sample project

**Comment by quickiwiki** [comment]
@JackBoosY please take a look https://github.com/quickiwiki/test_sqlite_orm

**Comment by JackBoosY** [comment]
@quickiwiki The repo you provided is not exists anymore, can you please provide it again?

Thanks.

**Comment by quickiwiki** [comment]
@JackBoosY Sorry, could you please try now?

**Comment by JackBoosY** [comment]
```
1>f:\vcpkg\installed\x86-windows\include\sqlite_orm\sqlite_orm.h(10146): fatal error C1001: An internal error has occurred in the compiler.
```

**Comment by JackBoosY** [comment]
Confirmed.

**Comment by JackBoosY** [comment]
No, this issue is not fixed in the latest master (commit ID: 115c6f64a340eac5bc5509bf884a7f350c9e6aa8).
Can you please report this to the upstream?

Thanks.
